File: TI1.2 SOAP_DOTNET_CONUNI_GR03/01 SERVIDOR/DOTNET_CONUNI_GRO3/Models/ConversionResponse.cs
using System.Runtime.Serialization;

namespace WCFService.Models
{
    [DataContract]
    public class CoordinateDMS
    {
        [DataMember]
        public int Degrees { get; set; }

        [DataMember]
        public int Minutes { get; set; }

        [DataMember]
        public double Seconds { get; set; }

        [DataMember]
        public char Hemisphere { get; set; }
    }

    [DataContract]
    public class ConversionResponse
    {
        // Latitude/Longitude in decimal degrees (echo)
        [DataMember]
        public double LatitudeDecimal { get; set; }

        [DataMember]
        public double LongitudeDecimal { get; set; }

        // Latitude/Longitude in radians
        [DataMember]
        public double LatitudeRadians { get; set; }

        [DataMember]
        public double LongitudeRadians { get; set; }

        // Latitude/Longitude in DMS
        [DataMember]
    public CoordinateDMS? LatitudeDMS { get; set; }

    [DataMember]
    public CoordinateDMS? LongitudeDMS { get; set; }

        // Mass conversions
        [DataMember]
        public double MassKg { get; set; }

        [DataMember]
        public double MassLb { get; set; }

        [DataMember]
        public double MassG { get; set; }
    }
}

File: TI1.2 SOAP_DOTNET_CONUNI_GR03/02 CLIWEB/WEBCLIENT_ConUni_SOAPDOTNET_GR03/src/main/java/ec/edu/pinza/webcli_conuni_soapdotnet_gr03/modelo/ConversionResponse.java
package ec.edu.pinza.webcli_conuni_soapdotnet_gr03.modelo;

public class ConversionResponse {

    private final Double massKg;
    private final Double massLb;
    private final Double massG;
    private final Double latitudeDecimal;
    private final Double latitudeRadians;
    private final CoordinateDMS latitudeDms;
    private final Double longitudeDecimal;
    private final Double longitudeRadians;
    private final CoordinateDMS longitudeDms;

    public ConversionResponse(
            Double massKg,
            Double massLb,
            Double massG,
            Double latitudeDecimal,
            Double latitudeRadians,
            CoordinateDMS latitudeDms,
            Double longitudeDecimal,
            Double longitudeRadians,
            CoordinateDMS longitudeDms
    ) {
        this.massKg = massKg;
        this.massLb = massLb;
        this.massG = massG;
        this.latitudeDecimal = latitudeDecimal;
        this.latitudeRadians = latitudeRadians;
        this.latitudeDms = latitudeDms;
        this.longitudeDecimal = longitudeDecimal;
        this.longitudeRadians = longitudeRadians;
        this.longitudeDms = longitudeDms;
    }

    public Double getMassKg() {
        return massKg;
    }

    public Double getMassLb() {
        return massLb;
    }

    public Double getMassG() {
        return massG;
    }

    public Double getLatitudeDecimal() {
        return latitudeDecimal;
    }

    public Double getLatitudeRadians() {
        return latitudeRadians;
    }

    public CoordinateDMS getLatitudeDms() {
        return latitudeDms;
    }

    public Double getLongitudeDecimal() {
        return longitudeDecimal;
    }

    public Double getLongitudeRadians() {
        return longitudeRadians;
    }

    public CoordinateDMS getLongitudeDms() {
        return longitudeDms;
    }
}

